subprojects/base-services/src/main/java/org/gradle/internal/FileUtils.java

```diff
@@ -173,12 +173,29 @@ private static boolean endsWithIgnoreCase(String subject, String suffix) {
      * @return the transformed path
      */
     public static String withExtension(String filePath, String extension) {
-        if (filePath.toLowerCase().endsWith(extension)) {
+        return withExtension(filePath, extension, new String[0]);
+    }
+
+    public static String withExtension(String filePath, String extension, String... alternativeExts) {
+        if (endsWithOneOf(filePath.toLowerCase(), extension, alternativeExts)) {
```

Comment:
At first I tried to implement this as follows:
```java
if (filePath.toLowerCase().endsWith(extension)
        || Arrays.stream(alternativeExts).anyMatch((altExt) -> filePath.toLowerCase().endsWith(extension))) {
```
but that resulted in an error that lambda is no supported with source level 6, so just to double-check I tried adding following to [`base-services.gradle.kts`](https://github.com/gradle/gradle/blob/master/subprojects/base-services/base-services.gradle.kts):
```kt
java {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
}
```
which resulted in a bunch of errors like this one when building the _base-services_ project using JDK 11.0.7:
```
D:\GIT\gradle\subprojects\base-services\src\main\java\org\gradle\api\specs\CompositeSpec.java:40: warning: [unchecked] Possible heap pollution from parameterized vararg type Spec<? super T>
    protected CompositeSpec(Spec<? super T>... specs) {
                                               ^
  where T is a type-variable:
    T extends Object declared in class CompositeSpec
error: warnings found and -Werror specified
```

Could anyone confirm that _base-services_ subproject is indeed built with Java source level 6 by design and I am not missing anything when it comes to building Gradle itself?